src/main/java/org/broadinstitute/hellbender/tools/spark/sv/Kmer.java

```diff
@@ -0,0 +1,177 @@
+    }
+
+    /**
+     * Returns a Kmer that is a canonical representation of this one.
+     * An odd-K Kmer is in canonical form if its middle base is A or C.
+     * The reverse-complement of a non-canonical Kmer is a canonical Kmer, and vice versa.  (Think about it.)
+     * Canonical form is not defined for even-K Kmers (too expensive to compute routinely).
+     */
+    public Kmer canonical( final int kSize )
+    {
+        if ( (kSize & 1) == 0 ) throw new IllegalArgumentException("K must be odd to canonicalize.");
+        if ( (v1 & 1L) == 0 ) return this;
+        return rc(kSize);
+    }
+
+    @Override
+    public boolean equals( final Object obj )
+    {
+        if ( !(obj instanceof Kmer) ) return false;
+        final Kmer that = (Kmer)obj;
+        return this.v1 == that.v1 && this.v2 == that.v2;
+    }
+
+    public boolean equals( final Kmer that )
+    {
+        return this.v1 == that.v1 && this.v2 == that.v2;
+    }
+
+    @Override
+    public int hashCode() { return (int)(v1 ^ (v1 >> 32) ^ v2 ^ (v2 >> 32)); }
+
+    /**
+     * Kmer comparison is consistent with equals.
+     * It's also the same as the lexicographic ordering you'd get using toString on the Kmers.
+     */
+    @Override
+    public int compareTo( final Kmer that )
+    {
+        int result = Long.compare(this.v1,that.v1);
+        if ( result == 0 ) result = Long.compare(this.v2,that.v2);
+        return result;
+    }
+
+    /**
+     * Not an override.  A Kmer doesn't know what K is, so it has to be supplied.
+     */
+    public String toString( final int kSize )
+    {
+        final StringBuilder sb = new StringBuilder(kSize);
+        long val = v2;
+        int nnn = kSize >> 1;
+        while ( nnn-- != 0 )
+        {
+            sb.append(baseChars[(int)val & 3]);
+            val >>= 2;
+        }
+        if ( (kSize & 1) == 0 )
+            val = v1;
+        else
+            val |= v1 << 1;
+        nnn = (kSize + 1) >> 1;
+        while ( nnn-- != 0 )
+        {
+            sb.append(baseChars[(int)val & 3]);
+            val >>= 2;
+        }
+        return sb.reverse().toString();
+    }
+
+
+    private Kmer( final long aV1, final long aV2 ) { v1 = aV1; v2 = aV2; }
+
+    // Reverse-complement a long by taking the reverse-complement of each of its bytes in reverse order.
+    private static long rcLong( long val )
+    {
+        long result = rcByte[(int)val & 0xFF];
+        int nBytes = 8;
+        while ( --nBytes != 0 )
+        {
+            val >>= 8;
+            result = (result << 8) | rcByte[(int)val & 0xFF];
+        }
+        return result;
+    }
+
+    private static final char baseChars[] = { 'A', 'C', 'G', 'T' };
+
+    // Lookup table for reverse-complementing each possible byte value.
+    // Each pair of bits represents a base, so you have to reverse bits pairwise and then invert all bits.
+    // This is most quickly and easily done with a lookup table.
+    private static final long rcByte[] =
```

Comment:
Yes please, it can be statically initialized.  Unless it takes more than a tiny fraction of second.